### PR TITLE
[2850] Popcorn wrapper defensive checks on ready media, and turn off cus...

### PR DESCRIPTION
**Pull request by ScottDowne**
...tom input keys while media is not ready.

### CODE REVIEW

File: src/core/popcorn-wrapper.js [hunk 1]
```diff
@@ -461,14 +464,14 @@ define( [ "core/logger", "core/eventmanager", "util/uri" ], function( Logger, Ev
 
     // Passthrough to the Popcorn instances play method
     this.play = function(){
-      if ( _popcorn.paused() ) {
+      if ( _popcorn.paused() && _mediaReady ) {
```
Comment (humphd) [inline src/core/popcorn-wrapper.js]: Flip the order of these calls, since we want to short-circuit if _mediaReady is false.

File: src/core/popcorn-wrapper.js [hunk 2]
```diff
@@ -461,14 +464,14 @@ define( [ "core/logger", "core/eventmanager", "util/uri" ], function( Logger, Ev
...
         _popcorn.play();
       }
     };
 
     // Passthrough to the Popcorn instances pause method
     this.pause = function(){
-      if ( !_popcorn.paused() ) {
+      if ( !_popcorn.paused() && _mediaReady ) {
```
Comment (humphd) [inline src/core/popcorn-wrapper.js]: Same here.

File: src/core/popcorn-wrapper.js [hunk 3]
```diff
@@ -572,7 +575,7 @@ define( [ "core/logger", "core/eventmanager", "util/uri" ], function( Logger, Ev
       currentTime: {
         enumerable: true,
         set: function( val ){
-          if( _popcorn ){
+          if( _popcorn && _mediaReady ){
```
Comment (humphd) [inline src/core/popcorn-wrapper.js]: and here

File: src/core/popcorn-wrapper.js (hunk 1 above)
Comment (ScottDowne) [inline src/core/popcorn-wrapper.js]: I did this in this order, because _popcorn.paused is more likely to be false than the media is to be not ready, thus we'll break sooner, more often.

That was my thinking, anyway.

I was also under the impression checking media.paused is safe when the media is not loaded, no?

Comment (humphd) [inline src/core/popcorn-wrapper.js]: Since _popcorn can be null, and you're doing a property lookup to get at paused(), it's safer to do the check on the boolean first.  I'd flip them, yes.

File: src/core/popcorn-wrapper.js [hunk 4]
```diff
@@ -609,10 +612,10 @@ define( [ "core/logger", "core/eventmanager", "util/uri" ], function( Logger, Ev
         set: function( val ){
           if( _popcorn ){
             if( val ){
-              _popcorn.pause();
```
Comment (humphd) [inline src/core/popcorn-wrapper.js:612]: Why this change?

File: src/core/popcorn-wrapper.js [hunk 5]
```diff
@@ -609,10 +612,10 @@ define( [ "core/logger", "core/eventmanager", "util/uri" ], function( Logger, Ev
...
+              _this.pause();
             }
             else {
-              _popcorn.play();
```
Comment (humphd) [inline src/core/popcorn-wrapper.js:615]: Same question here..
